### PR TITLE
Additional settings #1848

### LINKED ISSUE #1848: HastLayer: Add settings to disable some transforms
[HastLayer](https://hastlayer.com/) uses the decompiler library v2 and they want to move to the new decompiler engine. To make it easier for them, they want us to add some options to the `DecompilerSettings` class to tweak the decompilation process without having to rewrite some core transforms.

* Disable detection of all high-level loops except `while-true` and `while-condition` loops.
* An option to separate local variable declarations from their initializer.

/cc @Piedone

### PR DESCRIPTION
**Pull request by siegfriedpammer**
Add ForStatement, DoWhileStatement and SeparateLocalVariableDeclarations settings.

This fixes #1848

@dgrunwald please review my changes to the DeclareVariables transform. Thank you very much!

/cc @Piedone

### CODE REVIEW

**Comment by Piedone** [comment]
Thank you, I'll try it later today!

File: ICSharpCode.Decompiler/DecompilerSettings.cs [hunk 1]
```diff
@@ -1224,6 +1224,57 @@ public bool ArrayInitializers
 
 		#endregion
 
+		bool forStatement = true;
+
+		/// <summary>
+		/// Gets/sets whether the decompiler should produce for loops.
+		/// </summary>
+		[Category("DecompilerSettings.Other")]
```
Comment (dgrunwald) [inline ICSharpCode.Decompiler/DecompilerSettings.cs]: I would expect these language features in the C# 1.0 category, not in "Other".

**Comment by Piedone** [comment]
I've tried it out and the `SeparateLocalVariableDeclarations` config works great. I'm not sure about at least `ForStatement`: it being set to `false` if you decompile e.g. `ICSharpCode.ILSpy.Controls` then its ctor has a for loop, which will also be decompiled into a for loop. I'd expect this to be a while loop.

Am I misunderstanding something?

**Comment by siegfriedpammer** [comment]
@Piedone No, there was another transform producing for statements in the PatternStatementTransform, which I simply missed. Thank you for pointing that out! Should be fixed now...

**Comment by Piedone** [comment]
Great, thank you!

**Pull request merged**